File: frontend/src/types/category.ts
import { z } from "zod/v4"

export const categorySchema = z.object({
  id: z.string().uuid(),
  name: z.string().min(1),
  nameKey: z.string().optional(),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
})

export type Category = z.infer<typeof categorySchema>

export const categoryFormSchema = z.object({
  name: z.string().min(1),
})

export type CategoryFormData = z.infer<typeof categoryFormSchema>

File: backend/internal/model/category.go
package model

import (
	"errors"
	"time"

	"github.com/google/uuid"
)

type Category struct {
	ID        uuid.UUID `json:"id"`
	Name      string    `json:"name"`
	NameKey   string    `json:"nameKey,omitempty"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

type CategoryInput struct {
	Name string `json:"name"`
}

func (c *CategoryInput) Validate() error {
	if c.Name == "" {
		return errors.New("name is required")
	}
	return nil
}
